### PR TITLE
window/wayland: programmatic and interactive resize

### PR DESCRIPTION
**Pull request by UncleJ4ck**
adds runtime window resize.

`setSize(Vector2D)` sends an `xdg_surface.set_window_geometry` hint and commits. the compositor decides whether to honor it; `logicalSize` only updates from the `xdg_toplevel.configure` callback.

`startInteractiveResize(edge)` calls `xdg_toplevel.resize` using the latest pointer-button-PRESS serial so callers wiring this from a release-based click still pass a valid serial.

opt-in with `CWindowBuilder::resizable(true)`: the toolkit auto-checks the 6px edge zone in `mouseMove`, swaps the cursor to the right resize shape, and triggers `startInteractiveResize` on a left-click. element hit-test runs first so widgets pushed flush to the edge keep their clicks.

`eResizeEdge` mirrors xdg-shell edge values. cursor enum gains `HT_POINTER_RESIZE_NS/EW/NESW/NWSE`. no-op on layer, lock, popup.

### CODE REVIEW

File: src/window/WaylandWindow.cpp [hunk 1]
```diff
@@ -135,3 +135,103 @@ void CWaylandWindow::close() {
...
+    }
+    if (m_creationData.maxSize) {
+        clamped.x = std::min(clamped.x, m_creationData.maxSize->x);
+        clamped.y = std::min(clamped.y, m_creationData.maxSize->y);
+    }
+
+    // advisory hint to the compositor. don't mutate logicalSize here: the real
+    // size update arrives via the xdg_toplevel.configure callback, which is the
+    // only authoritative source for what the compositor agreed to.
+    m_waylandState.xdgSurface->sendSetWindowGeometry(0, 0, clamped.x, clamped.y);
```
Comment (vaxerski) [inline src/window/WaylandWindow.cpp]: this is not correct? Check the spec: https://wayland.app/protocols/xdg-shell#xdg_surface:request:set_window_geometry

I'm pretty sure the way that some clients try to do it is to either:
- soft: send a buffer of a different size (can't be done when maximized or protocol error)
- hard: send a min = max size at once

was this tested?

Comment (UncleJ4ck) [reply src/window/WaylandWindow.cpp]: yes it's tested but you are right it is not a resize request the compositor doesn't have to honor it as a size hint, i will dig into this

**Comment by UncleJ4ck** [comment]
@vaxerski follow-up: pushed two more commits.

`38c5ed6 window/wayland: pin min == max for setSize per spec` addresses your review. switched setSize from `set_window_geometry` to the min == max pin pattern. on the matching configure we restore the user's original min/max constraints so the window stays user-resizable afterwards.

`084b3b8 window/wayland: add autosize to track content size` adds a new builder flag. when `CWindowBuilder::autosize(true)` is set, the window follows `m_rootElement->preferredSize(0, 0)` on every reposition. passing zero as the parent makes PERCENT-sized descendants contribute nothing, so only ABSOLUTE and AUTO content shape the natural size. skipped during FULLSCREEN and RESIZING. MAXIMIZED is ignored because hyprland tags floating windows with it and we'd never autosize otherwise.

**Review by vaxerski — APPROVED** [review]
thanks! much better
